pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -158,26 +143,44 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 
 	ec2svc := ec2.NewService(scope.Scope)
 
-	clusterMachines, err := scope.MachineClient.List(v1.ListOptions{})
+	log.Info("Retrieving machines for cluster")
+	clusterMachines, err := scope.MachineClient.List(actuators.ListOptionsForCluster(cluster.Name))
 	if err != nil {
 		return errors.Wrapf(err, "failed to retrieve machines in cluster %q", cluster.Name)
 	}
 
 	controlPlaneMachines := GetControlPlaneMachines(clusterMachines)
+	if len(controlPlaneMachines) == 0 {
+		log.Info("No control plane machines exist yet - requeuing")
+		return &controllerError.RequeueAfterError{RequeueAfter: waitForControlPlaneMachineExistenceDuration}
+	}
+
+	// Except for the very first control plane machine, everything will be a join, so let's assume we're joining.
+	isNodeJoin := true
+	if cluster.Annotations[v1alpha1.AnnotationControlPlaneReady] != v1alpha1.ValueReady {
```

Comment:
Should we make this a method?

Reply:
Do you mean something to the effect of `determineIsNodeJoin`?

Reply:
Yeah something along those lines